File: src/Moq/Moq/contentFiles/vb/netstandard1.3/Mock.vb
Option Strict On

Imports System.Reflection
Imports Moq.Sdk

Namespace Global.Moq

    Partial Friend Class Mock

        Private Shared Function Create(Of T)(ByVal behavior As MockBehavior, ByVal constructorArgs As Object(), ParamArray interfaces As Type()) As T
            Dim mocked = DirectCast(MockFactory.[Default].CreateMock(GetType(Mock).GetTypeInfo().Assembly, GetType(T), interfaces, constructorArgs), IMocked)

            mocked.SetBehavior(behavior)

            Return DirectCast(mocked, T)
        End Function

    End Class

End Namespace

File: src/Moq/Moq/contentFiles/cs/netstandard1.3/Mock.cs
using System;
using System.Reflection;
using Moq.Sdk;

namespace Moq
{
    /// <summary>
    /// Instantiates mocks for the specified types.
    /// </summary>
    partial class Mock
    {
        /// <summary>
        /// Creates the mock instance by using the specified types to 
        /// lookup the mock type in the assembly defining this class.
        /// </summary>
        static T Create<T>(MockBehavior behavior, object[] constructorArgs, params Type[] interfaces)
        {
            var mocked = (IMocked)MockFactory.Default.CreateMock(typeof(Mock).GetTypeInfo().Assembly, typeof(T), interfaces, constructorArgs);

            mocked.SetBehavior(behavior);

            return (T)mocked;
        }
   }
}